source/robot_model/src/Model.cpp

```diff
@@ -294,12 +297,22 @@ state_representation::JointPositions Model::inverse_geometry(const state_represe
 
     q = pinocchio::integrate(this->robot_model_,q,v*dt);
 
+    for(uint j = 0; j < this->get_number_of_joints(); j++){
```

Comment:
don't wrap to M_PI - some real joints may support larger range of rotation (for example 720º joints). Other joints may not even be revolute (although at the moment we mainly think about revolute joints).

If you want to include this, then use the robot model joint limits directly.

Reply:
right, I put the wrap because it increments of a big delta_q (v*dt) and not with an infinitesimal value. But what I can do is to wrap this delta_q.